### ISSUE TITLE
Wrong servicebus resourceuri endpoint

### ISSUE THREAD

**Issue by serbrech** [Azure/go-autorest#565]
the resource uri defined in azure/environment.go is : 
https://github.com/Azure/go-autorest/blob/b076c1437d051bf4c328db428b70f4fe22ad38b0/autorest/azure/environments.go#L92

the servicebus sdk defines it as :
https://github.com/Azure/azure-service-bus-go/blob/c6cb351fc4abe639e1fc100b0d82040062a49fd0/namespace.go#L75

```
serviceBusResourceURI = "https://servicebus.azure.net/"
```

which one is the correct one?

Who should we tag? @jhendrixMSFT ? where is the documented source of truth?

**Comment by jhendrixMSFT** [comment]
Based on the docs [here](https://docs.microsoft.com/en-us/rest/api/servicebus/) I believe `servicebus.windows.net` is correct.

**Comment by serbrech** [comment]
so, it's the servicebus sdk that has a wrong url? but that works in our MSI tests though. do both work then?

**Comment by serbrech** [comment]
Oh, I think I mixed up Suffix (for service calls) and resourceURI (for authentication)

**Comment by serbrech** [comment]
is it simply missing from autorest then? 
```
ResourceIdentifiers: ResourceIdentifier{
			Graph:               "https://graph.windows.net/",
			KeyVault:            "https://vault.azure.net",
			Datalake:            "https://datalake.azure.net/",
			Batch:               "https://batch.core.windows.net/",
			OperationalInsights: "https://api.loganalytics.io",
			Storage:             "https://storage.azure.com/",
			Synapse:             "https://dev.azuresynapse.net",
		},
```

**Comment by jhendrixMSFT** [comment]
I missed this was for authentication :(.  Per [here](https://docs.microsoft.com/en-us/azure/service-bus-messaging/authenticate-application) you are correct, it's `https://servicebus.azure.net`

The list of `ResourceIdentifier` in `go-autorest` isn't comprehensive.  We started with a common base and expand it as customers request it.

I can add this for public cloud.  Looking for info on sovereign clouds.

**Comment by jhendrixMSFT** [comment]
Fixed in autorest/v0.11.7
